<?xml version="1.0"?>
<TeamMentor_Article Metadata_Hash="1505013209" Content_Hash="-517693959">
  <Metadata>
    <Id>26a68d55-95e2-4007-bfa6-d1eee30740de</Id>
    <Id_History>26a68d55-95e2-4007-bfa6-d1eee30740de,c00ebef7-6673-4d63-95a3-e469a74150c2,</Id_History>
    <Library_Id>be5273b1-d682-4361-99d9-6234f2d47eb7</Library_Id>
    <Title>Deallocation Functions Don't Throw Exceptions</Title>
    <Category>Error and Exception Management</Category>
    <Phase>Implementation</Phase>
    <Technology>Technology Independent</Technology>
    <Type>Checklist Item</Type>
    <DirectLink>Deallocation Functions Don't Throw Exceptions</DirectLink>
    <Tag />
    <Security_Demand />
    <Author />
    <Priority />
    <Status />
    <Source>SI</Source>
  </Metadata>
  <Content Sanitized="true" DataType="markdown">
    <Data><![CDATA[## What to Check For ##

Verify that deallocation functions don't throw exceptions.

## Why ##

Deallocation functions may be called when exceptions are thrown. If a deallocation function throws an exception after being called when an exception is thrown, the program may terminate. This may result in a denial-of-service vulnerability.

## How to Check ##

Search for deallocation functions. Verify that deallocation functions don't throw exceptions.

## How to Fix ##

Identify deallocation functions that throw exceptions and rewrite them in such a way that they don't throw exceptions.

## Additional Resources ##

 - For more information about not throwing exceptions from deallocation functions,  see [https://www.securecoding.cert.org/confluence/display/cplusplus/VOID+ERR38-CPP.+Deallocation+functions+must+not+throw+exceptions](https://www.securecoding.cert.org/confluence/display/cplusplus/VOID+ERR38-CPP.+Deallocation+functions+must+not+throw+exceptions)

]]></Data>
  </Content>
</TeamMentor_Article>